src/integrations/prefect-kubernetes/prefect_kubernetes/experimental/decorators.py

```diff
@@ -0,0 +1,173 @@
+        job_variables: dict[str, Any],
+        worker_cls: type[KubernetesWorker],
+    ) -> InfrastructureBoundFlow[P, R]:
+        new = cls(
+            flow.fn,
+            work_pool=work_pool,
+            job_variables=job_variables,
+            worker_cls=worker_cls,
+        )
+        # Copy all attributes from the original flow
+        for attr, value in flow.__dict__.items():
+            setattr(new, attr, value)
+        return new
+
+    @overload
+    def __call__(self: "Flow[P, NoReturn]", *args: P.args, **kwargs: P.kwargs) -> None:
+        # `NoReturn` matches if a type can't be inferred for the function which stops a
+        # sync function from matching the `Coroutine` overload
+        ...
+
+    @overload
+    def __call__(
+        self: "Flow[P, Coroutine[Any, Any, T]]",
+        *args: P.args,
+        **kwargs: P.kwargs,
+    ) -> Coroutine[Any, Any, T]: ...
+
+    @overload
+    def __call__(
+        self: "Flow[P, T]",
+        *args: P.args,
+        **kwargs: P.kwargs,
+    ) -> T: ...
+
+    @overload
+    def __call__(
+        self: "Flow[P, Coroutine[Any, Any, T]]",
+        *args: P.args,
+        return_state: Literal[True],
+        **kwargs: P.kwargs,
+    ) -> Awaitable[State[T]]: ...
+
+    @overload
+    def __call__(
+        self: "Flow[P, T]",
+        *args: P.args,
+        return_state: Literal[True],
+        **kwargs: P.kwargs,
+    ) -> State[T]: ...
+
+    def __call__(
+        self,
+        *args: "P.args",
+        return_state: bool = False,
+        wait_for: Optional[Iterable[PrefectFuture[Any]]] = None,
+        **kwargs: "P.kwargs",
+    ):
+        async def modified_call(
+            *args: P.args,
+            return_state: bool = False,
+            # TODO: Handle wait_for once we have an asynchronous way to wait for futures
+            wait_for: Optional[Iterable[PrefectFuture[Any]]] = None,
+            **kwargs: P.kwargs,
+        ) -> R | State[R]:
+            async with self.worker_cls(work_pool_name=self.work_pool) as worker:
+                parameters = get_call_parameters(self, args, kwargs)
+                future = await worker.submit(
+                    flow=self,
+                    parameters=parameters,
+                    job_variables=self.job_variables,
+                )
+                if return_state:
+                    await future.wait_async()
+                    return future.state
+                return await future.aresult()
+
+        if inspect.iscoroutinefunction(self.fn):
+            return modified_call(
+                *args, return_state=return_state, wait_for=wait_for, **kwargs
+            )
+        else:
+            return run_coro_as_sync(
+                modified_call(
+                    *args,
+                    return_state=return_state,
+                    wait_for=wait_for,
+                    **kwargs,
+                )
+            )
+
+
+def kubernetes(
+    work_pool: str, **job_variables: Any
```

Comment:
it'd be nice to have a `TypedDict` here to `Unpack` someday, not sure how we'd manage that, but just throwing it out there
```suggestion
    work_pool: str, **job_variables: Any
```

Reply:
I could see a world where we can generate a decorator for a specific work pool that includes the job variables as named and type kwargs. We're probably a ways off from that though.